src/mqtt_client/mod.rs

```diff
@@ -339,58 +415,89 @@ where
     {
         let settings = &mut self.settings;
         let mqtt = &mut self.mqtt;
-        let prefix = self.settings_prefix.as_str();
-
-        let mut response_topic: String<MAX_TOPIC_LENGTH> = String::from(self.prefix.as_str());
-        response_topic.push_str("/log").unwrap();
-        let default_response_topic = response_topic.as_str();
+        let prefix = self.prefix.as_str();
+        let listing_state = &mut self.listing_state;
 
         let mut updated = false;
         match mqtt.poll(|client, topic, message, properties| {
-            let path = match topic.strip_prefix(prefix) {
-                // For paths, we do not want to include the leading slash.
-                Some(path) => {
-                    if !path.is_empty() {
-                        &path[1..]
-                    } else {
-                        path
-                    }
-                }
-                None => {
-                    info!("Unexpected MQTT topic: {}", topic);
-                    return;
-                }
+            let Some(path) = topic.strip_prefix(prefix) else {
+                info!("Unexpected MQTT topic: {}", topic);
+                return;
+            };
+
+            let Ok(command) = Command::from_str(path) else {
+                info!("Unknown Miniconf command: {path}");
+                return;
             };
 
-            let mut new_settings = settings.clone();
-            let message: SettingsResponse = match new_settings.set(path, message) {
-                Ok(_) => {
-                    updated = true;
-                    handler(path, settings, &new_settings).into()
+            let mut data = [0u8; MESSAGE_SIZE];
+            let message = match command {
+                Command::Get => {
+                    let path = core::str::from_utf8(message).unwrap_or("");
+
+                    // Get a specific settings value with the provided path.
+                    settings
+                        .get(path, &mut data)
+                        .map(|len| &data[..len])
+                        .unwrap_or(b"Invalid path")
                 }
-                err => {
-                    let mut msg = String::new();
-                    if write!(&mut msg, "{:?}", err).is_err() {
-                        msg = String::from("Configuration Error");
+
+                Command::List => {
+                    if !properties
+                        .into_iter()
+                        .any(|prop| matches!(prop, Ok(minimq::Property::ResponseTopic(_))))
+                    {
+                        // If there's no response topic, there's no where we can publish the list.
+                        // Ignore the request.
+                        return;
```

Comment:
The fundamental issue here is that if the client provides no `ResponseTopic`, it's not clear where the response status should be published. Previously, we used a default `log` topic, which has since been removed.

Do we want to bring back a default topic?

Reply:
Ah no. Ignore and return here is fine.
My beef was with the return in L466 and https://github.com/quartiq/miniconf/pull/124/files#r1044566863